vllm/transformers_utils/tokenizers/mistral.py

```diff
@@ -391,144 +330,152 @@ def encode(
         max_length: Optional[int] = None,
         add_special_tokens: Optional[bool] = None,
     ) -> list[int]:
-        # `encode` should only be used for prompt completion
-        # it should never be used for chat_completion.
-        # For chat completion use `apply_chat_template`
         if add_special_tokens is not None:
-            return self.tokenizer.encode(
-                text, bos=add_special_tokens, eos=add_special_tokens
+            return self.transformers_tokenizer.encode(
+                text,
+                truncation=truncation,
+                max_length=max_length,
+                add_special_tokens=add_special_tokens,
             )
         else:
-            return self.tokenizer.encode(text, bos=True, eos=False)
+            encoded = self.tokenizer.encode(text, bos=True, eos=False)
+            if truncation is not None and not truncation:
+                return encoded
+            elif max_length is not None and len(encoded) > max_length:
+                return encoded[:max_length]
+            else:
+                return encoded
```

Comment:
```suggestion
            do_truncate = truncation and max_length is not None
            return encoded[:max_length] if do_truncate else encoded
```

Reply:
Ah I see, maybe just:

```py
truncation = truncation is not False and max_length is not None
```

should then do the trick. You can omit `len(encoded) > max_length` => it'll just be a no-op